contracts/oxo/OxoGame.sol

```diff
@@ -0,0 +1,137 @@
+Then game starts*/
+
+contract OxoGame is OxoConfig, GameEntity, GameAction, Players, Ownable {
+
+    event GameCreated(uint gameId, address creator, uint bet);
+    event GameStarted(uint gameId, address player);
+    event UserMadeMove(uint gameId, address player, uint8 index);
+    event GameWon(uint gameId, address winner);
+    event Draw(uint gameId);
+    event MoneyTransferred(uint gameId, address player, uint amount);
+    event GameCanceled(uint gameId);
+    event MoneyWithdrawn(address addr, uint amount);
+
+
+    function createGame() payable public returns (uint){
+        require(msg.value > 0 && msg.value <= maxBet);
+
+
+        Game storage game = createAndGetGame();
+        game.status = GameStatus.CREATED;
+        game.bet = msg.value;
+        game.gameId = games.length - 1;
+        addPlayer(game, msg.sender);
+
+        GameCreated(game.gameId, msg.sender, game.bet);
+        return game.gameId;
+    }
+
+    function joinGame(uint gameId) payable public returns (bool){
+        Game storage game = games[gameId];
+        require(game.status == GameStatus.CREATED);
+        require(msg.value == game.bet);
+        require(game.playerCount < maxPlayerCount);
+
+        addPlayer(game, msg.sender);
+        game.status = GameStatus.STARTED;
+        game.lastMove = msg.sender; // maybe a bit tricky. Game creator must move first
+
+        GameStarted(gameId, msg.sender);
+        return true;
+    }
+
+    function move(uint gameId, uint8 index) public returns (bool){
+        Game storage game = games[gameId];
+        require(index >= 1 && index <= maxFieldNumber);
+        require(game.status == GameStatus.STARTED);
+        require(isPlayer(game, msg.sender)); // if this player has already joined the game
+        require(game.lastMove != msg.sender); // prevent user to make two moves successively
+        require(!isMoved(game, index)); // is move unique
+
+        addMove(game, index, msg.sender);
+        UserMadeMove(gameId, msg.sender, index);
+
+        if (games[gameId].moveCount[msg.sender] <= 2) return true; // if player made only 2 moves he cannot lose or win already
+        bool result = verifyMove(game);
+        bool isLastMove = game.allMoveCount == maxFieldNumber;
+        if (result || isLastMove) {
+            if (result) {
+                game.winner = msg.sender;
+                GameWon(gameId, msg.sender);
+            } else if (isLastMove) Draw(gameId);
+            finishGame(game);
+        }
+        return true;
+    }
+
+
+    function cancelGame(uint gameId) public returns (bool){
+
+        Game storage game = games[gameId];
+        require(game.status != GameStatus.FINISHED);
+        require(isPlayer(game, msg.sender));
+        require(!isCanceler(game, msg.sender));//user cannot cancel game more than once
+
+        addCanceler(game, msg.sender);
+        //players can cancel the game if they both decided this or if nobody wants to play with player one :(
```

Comment:
What will happen if second player has joined to game but after several moves understood he will loose and stop playing. And doesn't cancel game. Will first player loose his money?

Reply:
congratulate u. u have found a bug... 
do not even know how to correctly fix it... 
write game analyzer  that can check who has won is too expensive, IMO. or i can be wrong...
another case  to allow user withdraw the money if after some time... but it is also not perfect way

Reply:
>another case to allow user withdraw the money after some time. 
It will work